src/main/scala/com.socrata.tileserver/util/RenderProvider.scala

```diff
@@ -27,18 +32,21 @@ import RenderProvider._
 case class RenderProvider(http: HttpClient, baseUrl: RequestBuilder) {
   /** Render the provided tile using the provided request info.
     *
-    * @param rawTile a Map that contains the features as WKB.
+    * @param encoder An encoder to manipulate features
     * @param info the request info to use while rendering the tile.
     */
-  def renderPng(rawTile: MapTile, info: RequestInfo): InputStream = {
-    val style = info.style.get
-    val tile: Map[String, Seq[String]] = rawTile.map { case (layer, wkbs) =>
-      layer -> wkbs.map(Base64.encodeBase64String(_))
+  def renderPng(encoder: TileEncoder, info: RequestInfo): InputStream = {
+    val rawTile = encoder.wkbsAndAttributes
+    val cartoCssProvider = CartoCssEncoder(info)
+    val style  = (info.min, info.max) match {
+      case (Some(_), Some(_)) => cartoCssProvider.buildCartoCSS
+      case _ => info.style.get
     }
 
-    val content: Map[String, Any] = Map("tile" -> tile,
+
+    val content: Map[String, Any] = Map("tile" -> rawTile,
                                         "zoom" -> info.zoom,
-                                        "style" -> style,
+                                        "style" -> Base64.encodeBase64String(style.getBytes),
```

Comment:
This isn't the correct way go to go about that, it sounds like you maybe did it to reduce payload size?

In that case you should probably actually compress the file instead of base64 encoding it.